enhancements/olm/recursive-cluster-operators.md

```diff
@@ -0,0 +1,415 @@
+That includes information about underlying ReplicaSets, which the Deployment controller manages on behalf of the Deployment, and recursively from Pods, which the ReplicaSet controller manages on behalf of the ReplicaSets.
+But it does not include information from an operand branching out to manage something directly, like the etcd operator managing etcd members.
+The Deployment controller has no idea what's going on with etcd members.
+
+For all of OpenShift v4, we have had a a [ClusterOperator type][ClusterOperator].
+This gives operators who manage in-cluster resources a pathway in parallel with their own Deployment to reach around the Deployment controller and pass information directly back to the CVO.
+For example, the etcd operator can say "etcd members and my other operands are healthy, and we're currently all at 4.9.99".
+Between the etcd ClusterOperator `status` and Deployment `status`, the CVO has a complete picture of how the etcd operator is doing.
+
+The CVO does not have a monopoly on having operators as its operands.
+For example, OLM is is the business of installing and managing operators.
+Something installs and manages the Deployment operator.
+The monitoring operator installs and manages a Prometheus operator.
+All of these operators need a way to get information back out of their operands about how their operating is going, so they can in turn pass that information up to their parents.
+The ClusterOperator type is a very generic `status` object, and would be a good vessel for transmitting this information.
+
+### Tracking ownership
+
+No need for a `spec.parent` property or anything; we already have `metadata.ownerReferences`.
+Might need a namespaced variant?  I dunno how many operators can actually be installed with the same name in parallel namespaces.
+
+### Upstream
+
+If we are pitching this as "hey, this is a fundamental gap", it makes sense to take it upstream.
+I have no idea what's upstream in this space; maybe there's already something there.
+
+### User Stories
+
+#### OLM
+
+OLM starts creating ClusterOperator for each of its operand operators, just like the CVO does today.
+OLMs operand operators manage whatever it is they manage, and report on status up to OLM via those ClusterOperators.
+OLM watches Deployments, ClusterOperators, etc., and reports on its status up to the CVO via its own ClusterOperator.
+Console writes a really simple UX that allows admins to browse their whole cluster's component tree:
+
+* Cluster level.  OLM sounds interesting.  Click.
+* OLM level.  ExampleOpA sounds interesting.  Click.
+* ExampleOpA level.  ExampleOpB sounds interesting.  Click.
+* ...
+
+Life is great.
+
+#### CVO
+
+CVO probably dogfoods this pattern by growing its own ClusterOperator where it sets `conditions` and things.
```

Comment:
This implies a CO without an owner or where the owner is itself.  Moving conditions out of ClusterVersion will have impacts on a lot of tooling.  I don't see the benefit in this change.

Reply:
`ownerReferences` only works when you're referencing in-cluster owners.  It can't point at human admins.  It probably can't even point across the HyperShift control-plane/managed-compute boundary.  So the `ownerReferences` there do need to remain empty or self-referential as a marker for "there is no in-cluster owner for this".  Benefit of the self-referential `ownerReferences` is that it is clearly distinct from "`ownerReferences` is empty because the owner didn't think to set it".

And I agree about the impact of dropping `conditions` from ClusterVersion, which is why I say "but backwards compat means that will probably never happen" a few lines below this sentence.